apps/designer/app/designer/features/style-panel/sections/layout/layout.tsx

```diff
@@ -1,13 +1,239 @@
-import { Box, Flex, Grid } from "@webstudio-is/design-system";
+import { useState } from "react";
+import {
+  Box,
+  DeprecatedIconButton,
+  EnhancedTooltip,
+  Flex,
+  Grid,
+  Tooltip,
+} from "@webstudio-is/design-system";
+import type { StyleProperty, StyleValue } from "@webstudio-is/css-data";
 import { toValue } from "@webstudio-is/css-engine";
+import {
+  ColumnGapIcon,
+  RowGapIcon,
+  LinkedIcon,
+  UnlinkedIcon,
+} from "@webstudio-is/icons";
 import type { RenderCategoryProps } from "../../style-sections";
 import { FlexGrid } from "./shared/flex-grid";
-import { Lock } from "./shared/lock";
 import { renderProperty } from "../../style-sections";
-import { MenuControl, SelectControl, TextControl } from "../../controls";
+import { MenuControl, SelectControl } from "../../controls";
 import { PropertyName } from "../../shared/property-name";
-import { ColumnGapIcon, RowGapIcon } from "@webstudio-is/icons";
 import { styleConfigByName } from "../../shared/configs";
+import type { CreateBatchUpdate } from "../../shared/use-style-data";
+import { getStyleSource, type StyleInfo } from "../../shared/style-info";
+import {
+  type IntermediateStyleValue,
+  CssValueInput,
+} from "../../shared/css-value-input";
+
+const GapLinked = ({
+  isLinked,
+  onChange,
+}: {
+  isLinked: boolean;
+  onChange: (isLinked: boolean) => void;
+}) => {
+  return (
+    <Tooltip
+      content={isLinked ? "Unlink gap values" : "Link gap values"}
+      delayDuration={400}
+      disableHoverableContent={true}
+    >
+      <Flex
+        css={{
+          width: "100%",
+          justifyContent: "center",
+        }}
+      >
+        <DeprecatedIconButton onClick={() => onChange(!isLinked)}>
```

Comment:
please don't use !i they are really hard to see in most fonts

Reply:
Agree, don't use it most of the time. Find it useful only for toggling booleans. But it's just a visual habit.